File: algorithm/python/BOJ11399.py
# problem : https://www.acmicpc.net/problem/11399

import sys

N = int(sys.stdin.readline().rstrip())
time_list = list(map(int, sys.stdin.readline().rstrip().split()))

time_list.sort()

answer = 0
for i in range(0, N):
    for j in range(0, i + 1):
        answer += time_list[j]

print(answer)

File: algorithm/typescript/BOJ11399.ts
// problem : https://www.acmicpc.net/problem/11399

import fs = require("fs");

const input: string[] = fs.readFileSync("/dev/stdin").toString().split("\n");
const N: number = Number(input[0]);
const list: number[] = input[1].split(" ").map(Number);

list.sort((a, b) => {
  return a - b;
});

let answer: number = 0;
for (let i = 0; i < N; i++) {
  for (let j = 0; j <= i; j++) {
    answer += list[j];
  }
}

console.log(answer);
